doc/specifications/rest/REST-API-V2.rst

```diff
@@ -111,7 +121,38 @@ In this document, for the sake of readability, no prefix is used in the URIs. In
 prefixes all REST hrefs.
 
 Remember that URIs to REST resources should never be generated manually, but obtained from earlier REST
- calls.
+calls.
+
+Alias URIs
+----------
+
+By default entity URIs are constructed using a (server local) URL template and
+the database ID of the referenced entity. While this mechanism might be subject
+to change (e.g. to support multi-server/-domain setups with interlinking)
+users are requesting more intuitive ways to identify entities to reference
+entities while creating/updating related entities. Alias URIs help to achieve
+this by providing more human readable resource identifiers. These URIs only
+support GET/HEAD operations and will return a redirect to the cannonical entity
+URL. However, when using them as a reference in a POST/PUT/PATCH request, the
+server will accept them and automatically reference the correct entity.
+
+Alias URLs are built using the collection resource of an entity + a filter
+query parameter which identifies the entity in question uniquely (e.g.
+``remoteId`` or ``identifier``).
+
+A request/response cycle for an alias URI is supposed to be executed as follows:
+
+:Resource: /content/types/?identifier=article
+:Method: GET
+:Response:
+
+.. code:: http
+
+        HTTP/1.1 307 Temporary Redirect
+        Location: /content/types/23
```

Comment:
Note to self: we can cache these with Vanrish: http://book.varnish-software.com/4.0/chapters/VCL_Basics.html#the-initial-value-of-beresp-ttl

Assuming we also return cache headers and proper tag header so ezplatform-http-cache can invalidate it _(needed if we cache this)_.